plugins/auto-sizes/hooks.php

```diff
@@ -63,12 +63,12 @@ function auto_sizes_update_content_img_tag( $html ): string {
 	}
 
 	// Bail early if the image is not responsive.
-	if ( false === strpos( $html, 'sizes="' ) ) {
+	if ( preg_match( '/sizes="([^"]+)"/', $html, $match ) === 0 || ! isset( $match[1] ) ) {
```

Comment:
This is simpler and with it PHPStan is smart enough to know that `$match[1]` is set:

```suggestion
	if ( 1 !== preg_match( '/sizes="([^"]+)"/', $html, $match ) ) {
```

Reply:
@felixarntz Any issue with this suggestion?

Reply:
Had missed it :)

Reply:
Have you checked though that PHPStan is fine with it? It was requiring me to check for `isset( $match[1] )`, which I really added just to please the tool. :)

Reply:
Yes, I tested locally.

Reply:
couple quick thoughts:
 - there's no need to store the result or pass in `$match` since we're not using it
 - the `sizes` attribute could be in any case, but this only tests for all-lowercase matches
 - the `sizes` attribute could have single quotes, or no quotes, and have whitespace around the equals sign, e.g. `Sizes = Auto`
 - the `sizes` attribute _must_ be preceded by whitespace or a `/`, but this test also matches `data-sizes="unrelated attribute"`

Reply:
With the current code, PHPStan types `$match` as `array{0?: string, 1?: non-empty-string}`

But with the suggested change, the type becomes `array{string, non-empty-string}`.

Reply:
@dmsnell:

> * there's no need to store the result or pass in `$match` since we're not using it

Further below the `$match` is used:

```php
auto_sizes_attribute_includes_valid_auto( $match[1] )
```

> * the `sizes` attribute could be in any case, but this only tests for all-lowercase matches
> * the `sizes` attribute could have single quotes, or no quotes, and have whitespace around the equals sign, e.g. `Sizes = Auto`
> * the `sizes` attribute _must_ be preceded by whitespace or a `/`, but this test also matches `data-sizes="unrelated attribute"`

Good points. These are tracked in https://github.com/WordPress/performance/issues/1446.

Reply:
ah. well I got lost in the regexes and the attempt to parse HTML with them 😉

`$size_attribute = $processor->get_attribute( 'sizes' );` does have a certain ring to it.